mcp_client.py

```diff
@@ -0,0 +1,152 @@
+import asyncio
+import json
+import os
+from contextlib import AsyncExitStack
+from typing import Dict, List, Optional, TypeAlias
+
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+
+from huggingface_hub import AsyncInferenceClient, ChatCompletionInputTool, ChatCompletionOutput
+from huggingface_hub.inference._providers import PROVIDER_T
+
+
+# Type alias for tool names
+ToolName: TypeAlias = str
+
+
+class MCPClient:
```

Comment:
could potentially also support 2 syntaxes like in the MCPClient for smolagents:
1. like you have with try and finally
2. one as a context manager, where you directly pass the the MCP client settings + the MCP servers with the tools you want to have this would allow to do something like "with MCPClient(...) as client:" and then just run client.chat.completion with the tools